### ISSUE TITLE
Support `shimmed` package

### ISSUE THREAD

**Issue by SBoudrias** [caolan/jam#111]
Hi,

There's a lot of package manager out there (volo, bower, etc) and as so, sometime authors don't want to add configuration for each one into their project.

It would be great if we could shim package from github repo without waiting for a project author to add jam config to it's `package.json`.

For example, shimmed package could work as so: on our local project, we add a .jam file (or anything) and inside we define the way packages should be fetch.

**.jam**

``` json
{
    "shim_packages": {
        "handlebars": {
            "url"      : "https://github.com/documentcloud/backbone",
            "main" : "./backbone.js",
            "dependencies": {
                "jquery"           : ">1.4.2",
                "underscore" : ">1.3.1"
            },
            "shim": {
                "deps"      : ["jquery"],
                "exports"  : "Example"
            }
        }
    }
}
```

Basically, inside a shimmed package definition, we just add what could be the content of a `package.json` file for the particular repo.

Then, we would only need to type `jam install backbone` in the terminal.

What do you think?

---

I saw that some packages already seems to have been added through some shimmed way ([Handlebars](http://jamjs.org/packages/#/details/handlebars), [Backbone](http://jamjs.org/packages/#/details/backbone), etc), but haven't find any documentation at this subject. Is there any?
